From 44e1631fb04f45ff6eec0f7d82b456b37caa0ca1 Mon Sep 17 00:00:00 2001
From: Zhou Peng <eagle.zhou@nxp.com>
Date: Fri, 31 Aug 2018 14:18:24 +0800
Subject: [PATCH 4500/5242] MLK-19398 - [i.MX8MM/Hantro] Hantro library crash
 when doing suspend and resume stress test

commit  96b5348b177642bb04d178aa1b91ccfb17af5239 from
https://source.codeaurora.org/external/imx/linux-imx.git

Replace wait_event_interruptible_timeout()/wake_up_interruptible_all() with wait_event_timeout()/wake_up_all()
So one complete decode process will be guranteed to avoid potential assert be triggered.

Signed-off-by: Zhou Peng <eagle.zhou@nxp.com>
(cherry picked from commit 64634b90c6f6b15a97b9cae91b0e6099970cee55)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/hantro_845/hantrodec_845s.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/mxc/hantro_845/hantrodec_845s.c b/drivers/mxc/hantro_845/hantrodec_845s.c
index f6be601..c151995 100755
--- a/drivers/mxc/hantro_845/hantrodec_845s.c
+++ b/drivers/mxc/hantro_845/hantrodec_845s.c
@@ -870,7 +870,8 @@ static long WaitDecReadyAndRefreshRegs(hantrodec_t *dev, struct core_desc *Core)
 
 	PDEBUG("wait_event_interruptible DEC[%d]\n", dev->core_id);
 
-	ret = wait_event_interruptible_timeout(dec_wait_queue, CheckDecIrq(dev), msecs_to_jiffies(200));
+	//ret = wait_event_interruptible_timeout(dec_wait_queue, CheckDecIrq(dev), msecs_to_jiffies(200));
+	ret = wait_event_timeout(dec_wait_queue, CheckDecIrq(dev), msecs_to_jiffies(200));
 	if (ret == -ERESTARTSYS) {
 		pr_err("DEC[%d]  failed to wait_event_interruptible interrupted\n", dev->core_id);
 		return -ERESTARTSYS;
@@ -1779,7 +1780,8 @@ static irqreturn_t hantrodec_isr(int irq, void *dev_id)
 
 			dec_irq |= (1 << dev->core_id);
 
-			wake_up_interruptible_all(&dec_wait_queue);
+			//wake_up_interruptible_all(&dec_wait_queue);
+			wake_up_all(&dec_wait_queue);
 			handled++;
 		}
 	//}
-- 
1.7.9.5

